# Tugas3 Matriks invers

1. sifat-sifat matrix invers

2. mencari invers dengan eliminasi gaus jordan

3. solusi sistem persamaan linier dengan matrix invers

4. (membuat soal persamaan linier 6 variabel)

# pengertian Matrix Invers

Dalam aljabar linear matrix persegi A n x n disebut invertible (atau juga disebut nonsingular, nondegenerate, atau biasanya digunakan secara umum).

Jika terdapat sebuah matrix persegi B n x n maka: AB = BA = I, dimana I adalah dinotasikan dengan matrix Identitas n x n dan perkalian yang digunakan adalah perkalian matrix biasa. Maka Matrix B secara unik ditentukan oleh Matrix A, dan disebut Inverse dari Matrix A. dinotasikan dengan

. Matrix inversion adalah proses untuk menemukan invers marix dari invertible Matrix. Lain halnya Matrix persegi yang tidak dapat dibalik disebut Matrix singular atau matrix tunggal atau degenerate

# Sifat sifat Matrix Invers

Matrix invers adalah sifat yang berlaku pada matrix yang merupakan kebalikan dari Matrix asal. Matrix invers dapat ditulis dengan simbol yang berarti jika dikalikan dengan Matrix A akan menghasilkan Matrix Identitas atau Matrix. Matrix Identitas adalah Matrix yang memiliki angka 1 sebagai diagonal utamanya dan selain itu angka 0 nilainya.

**Berikut adalah sifat sifat Matrix Invers**

1. Matrix Invers hanya dimiliki oleh Matrix persegi atau Matrix yang memiliki baris dan kolom yang sama

2. Matrix invers hanya ada pada Matrix yang non-singular, yaitu Determinannya tidak sama dengan 0

3. Invers Matrix bersifat tunggal, artinya tidak ada matrix invers selain yang menjadi invers dari matrix tersebut. Jika B dan C adalah invers dari Matrix A, maka B = C.

4. Invers dari invers suatu Matrix adalah Matrix itu sendiri

5. Invers dari hasil kali dua atau lebih matriks adalah hasil kali invers-inversnya dalam urutan terbalik. Jika A dan B keduanya nonsingular, maka (AB)^-1 = B^-1 A^-1.

# Mencari invers dengan eliminasi gaus jordan

Dalam penerapan gauss Jordan dengan menggunakan Operasi Baris Elementer atau disebut OBE. Operai OBE terdiri dari 3 cara, yaitu:

Menambahkan kelipatan baris ke baris yang lain

mengkalikan baris dengan bilangan selain nol

Menukarkan posisi baris satu dengan baris yang lain

Untuk mencari Invers dari sebuah Matrix kita harus memperhatikan persamaan - persamaannya. Untuk bisa memasukkannya dalam bentuk matrix kita pisah antara koefisien yang ada disebelah ruas kiri dan konstanta yang ada disebelah ruas kanan. contoh saya mempunyai persamaan linier 6 variabel:

2X_1 + 2X_2 + 2X_3 + 4X_4 + 3X_5 + 6X_6 = 25

2X_1 + 2X_2 + X_3 + 4X_4 + 4X_5 + 5X_6 = 22

X_1 + X_2 + 3X_3 + 4X_4 + X_5 + 6X_6 = 23

3X_1 + 2X_2 + X_3 + X_4 + 2X_5 + 3X_6 = 16

2X_1 + X_2 + 2X_3 + 3X_4 + X_5 + 3X_6 = 17

X_1 + 2X_2 + X_3 + X_4 + 3X_5 + X_6 = 11

Lalu kita akan masukkan dalam bentuk matrix:

\begin{align*} KoefisienA= \begin{bmatrix} 2&2&2&4&3&6\ 2&-2&1&4&4&5\ 1&1&3&4&1&6\ 3&2&1&1&2&3\ 2&1&2&3&1&3\ 1&2&1&1&3&1 \end{bmatrix} \ KonstanA= \begin{bmatrix} 25\ 22\ 23\ 16\ 17\ 13 \end{bmatrix} \ I= \begin{bmatrix} 1&0&0&0&0&0\ 0&1&0&0&0&0\ 0&0&1&0&0&0\ 0&0&0&1&0&0\ 0&0&0&0&1&0\ 0&0&0&0&0&1 \end{bmatrix} \end{align*}

# Solusi sistem persamaan linier dengan matrix invers dengan 6 variabel

Untuk bisa kita menemukan invers dari suatu matrix, kita harus melakukan operasi - operasi baris elementer pada Matrix Identitas yang dimana operasi operasi baris elemternya sama dengan operasi - operasi baris elementer yang mengubah Matrix KoefisienA menjadi matrix identitas

# Penyelesaian dengan code

Berikut adalah contoh penyelesaian mencari Invers Matrix 6x6 dengan code python

In [1]:
# Import library numpy and memasukkan persamaan dalam bentuk matrix
import numpy as np
A = np.array([[2,2,2,4,3,6,25],[2,2,1,4,4,5,22],[1,1,3,4,1,6,23],[3,2,1,1,2,3,16],[2,1,2,3,1,3,17],[1,2,1,1,3,1,13]])
print(f"Matrix A  = \n{A}")
print("\n")
B  = np.array([[1,0,0,0,0,0],[0,1,0,0,0,0],[0,0,1,0,0,0],[0,0,0,1,0,0],[0,0,0,0,1,0],[0,0,0,0,0,1]])
print(f"Matrix I = \n{B}")

Matrix A  = 
[[ 2  2  2  4  3  6 25]
 [ 2  2  1  4  4  5 22]
 [ 1  1  3  4  1  6 23]
 [ 3  2  1  1  2  3 16]
 [ 2  1  2  3  1  3 17]
 [ 1  2  1  1  3  1 13]]


Matrix I = 
[[1 0 0 0 0 0]
 [0 1 0 0 0 0]
 [0 0 1 0 0 0]
 [0 0 0 1 0 0]
 [0 0 0 0 1 0]
 [0 0 0 0 0 1]]


In [5]:
def RowSwap(A,k,l):
# =============================================================================
#     A is a NumPy array.  RowSwap will return duplicate array with rows
#     k and l swapped. (menukarkan persamaan)
# =============================================================================
    m = A.shape[0]  # m is number of rows in A (jumlah baris A)
    n = A.shape[1]  # n is number of columns in A (jumlah kolom A)

    B = np.copy(A).astype('float64')

    for j in range(n):
        temp = B[k][j]
        B[k][j] = B[l][j]
        B[l][j] = temp

    return B

def RowScale(A,k,scale):
# =============================================================================
#     A is a NumPy array.  RowScale will return duplicate array with the
#     entries of row k multiplied by scale. (mengalikan entry baris A dengan skala)
# =============================================================================
    m = A.shape[0]  # m is number of rows in A
    n = A.shape[1]  # n is number of columns in A

    B = np.copy(A).astype('float64')

    for j in range(n):
        B[k][j] *= scale

    return B

def RowAdd(A,k,l,scale):
# =============================================================================
#     A is a numpy array.  RowAdd will return duplicate array with row
#     l modified.  The new values will be the old values of row l added to
#     the values of row k, multiplied by scale. (menambahkan nilai baris k dikalikan dengan skala)
# =============================================================================
    m = A.shape[0]  # m is number of rows in A
    n = A.shape[1]  # n is number of columns in A

    B = np.copy(A).astype('float64')

    for j in range(n):
        B[l][j] += B[k][j]*scale

    return B

In [6]:
A1 = RowSwap(A,0,2)
print(A1)
print("\n")

B1 = RowSwap(B,0,2)
print(B1)
print("\n")

A2 = RowAdd(A1,0,1,-2)
print(A2)
print("\n")

B2 = RowAdd(B1,0,1,-2)
print(B2)
print("\n")

A3 = RowAdd(A2,0,2,-2)
print(A3)
print("\n")

B3 = RowAdd(B2,0,2,-2)
print(B3)
print("\n")

A4 = RowAdd(A3,0,3,-3)
print(A4)
print("\n")

B4 = RowAdd(B3,0,3,-3)
print(B4)
print("\n")

A5 = RowAdd(A4,0,4,-2)
print(A5)
print("\n")

B5 = RowAdd(B4,0,4,-2)
print(B5)
print("\n")

A6 = RowAdd(A5,0,5,-1)
print(A6)
print("\n")

B6 = RowAdd(B5,0,5,-1)
print(B6)

[[ 1.  1.  3.  4.  1.  6. 23.]
 [ 2.  2.  1.  4.  4.  5. 22.]
 [ 2.  2.  2.  4.  3.  6. 25.]
 [ 3.  2.  1.  1.  2.  3. 16.]
 [ 2.  1.  2.  3.  1.  3. 17.]
 [ 1.  2.  1.  1.  3.  1. 13.]]


[[0. 0. 1. 0. 0. 0.]
 [0. 1. 0. 0. 0. 0.]
 [1. 0. 0. 0. 0. 0.]
 [0. 0. 0. 1. 0. 0.]
 [0. 0. 0. 0. 1. 0.]
 [0. 0. 0. 0. 0. 1.]]


[[  1.   1.   3.   4.   1.   6.  23.]
 [  0.   0.  -5.  -4.   2.  -7. -24.]
 [  2.   2.   2.   4.   3.   6.  25.]
 [  3.   2.   1.   1.   2.   3.  16.]
 [  2.   1.   2.   3.   1.   3.  17.]
 [  1.   2.   1.   1.   3.   1.  13.]]


[[ 0.  0.  1.  0.  0.  0.]
 [ 0.  1. -2.  0.  0.  0.]
 [ 1.  0.  0.  0.  0.  0.]
 [ 0.  0.  0.  1.  0.  0.]
 [ 0.  0.  0.  0.  1.  0.]
 [ 0.  0.  0.  0.  0.  1.]]


[[  1.   1.   3.   4.   1.   6.  23.]
 [  0.   0.  -5.  -4.   2.  -7. -24.]
 [  0.   0.  -4.  -4.   1.  -6. -21.]
 [  3.   2.   1.   1.   2.   3.  16.]
 [  2.   1.   2.   3.   1.   3.  17.]
 [  1.   2.   1.   1.   3.   1.  13.]]


[[ 0.  0.  1.  0.  0.  0.]
 [ 0.  1. -2.  0.  0.  0.]
 

In [7]:
A7 = RowSwap(A6,5,1)
print(A7)
print("\n")

B7 = RowSwap(B6,5,1)
print(B7)
print("\n")

A8 = RowAdd(A7,1,0,-1)
print(A8)
print("\n")

B8 = RowAdd(B7,1,0,-1)
print(B8)
print("\n")

A9 = RowAdd(A8,1,3,1)
print(A9)
print("\n")

B9 = RowAdd(B8,1,3,1)
print(B9)
print("\n")

A10 = RowAdd(A9,1,4,1)
print(A10)
print("\n")

B10 = RowAdd(B9,1,4,1)
print(B10)
print("\n")

[[  1.   1.   3.   4.   1.   6.  23.]
 [  0.   1.  -2.  -3.   2.  -5. -10.]
 [  0.   0.  -4.  -4.   1.  -6. -21.]
 [  0.  -1.  -8. -11.  -1. -15. -53.]
 [  0.  -1.  -4.  -5.  -1.  -9. -29.]
 [  0.   0.  -5.  -4.   2.  -7. -24.]]


[[ 0.  0.  1.  0.  0.  0.]
 [ 0.  0. -1.  0.  0.  1.]
 [ 1.  0. -2.  0.  0.  0.]
 [ 0.  0. -3.  1.  0.  0.]
 [ 0.  0. -2.  0.  1.  0.]
 [ 0.  1. -2.  0.  0.  0.]]


[[  1.   0.   5.   7.  -1.  11.  33.]
 [  0.   1.  -2.  -3.   2.  -5. -10.]
 [  0.   0.  -4.  -4.   1.  -6. -21.]
 [  0.  -1.  -8. -11.  -1. -15. -53.]
 [  0.  -1.  -4.  -5.  -1.  -9. -29.]
 [  0.   0.  -5.  -4.   2.  -7. -24.]]


[[ 0.  0.  2.  0.  0. -1.]
 [ 0.  0. -1.  0.  0.  1.]
 [ 1.  0. -2.  0.  0.  0.]
 [ 0.  0. -3.  1.  0.  0.]
 [ 0.  0. -2.  0.  1.  0.]
 [ 0.  1. -2.  0.  0.  0.]]


[[  1.   0.   5.   7.  -1.  11.  33.]
 [  0.   1.  -2.  -3.   2.  -5. -10.]
 [  0.   0.  -4.  -4.   1.  -6. -21.]
 [  0.   0. -10. -14.   1. -20. -63.]
 [  0.  -1.  -4.  -5.  -1.  -9. -29.]
 [  0.   0.  -5.  

In [8]:
A11 = RowAdd(A10,5,2,-1)
print(A11)
print("\n")

B11 = RowAdd(B10,5,2,-1)
print(B11)
print("\n")

A12 = RowAdd(A11,2,0,-5)
print(A12)
print("\n")

B12 = RowAdd(B11,2,0,-5)
print(B12)
print("\n")

A13 = RowAdd(A12,2,1,2)
print(A13)
print("\n")

B13 = RowAdd(B12,2,1,2)
print(B13)
print("\n")

A14 = RowAdd(A13,2,3,10)
print(A14)
print("\n")

B14 = RowAdd(B13,2,3,10)
print(B14)
print("\n")

A15 = RowAdd(A14,2,4,6)
print(A14)
print("\n")

B15 = RowAdd(B14,2,4,6)
print(B14)
print("\n")

A16 = RowAdd(A15,2,5,5)
print(A16)
print("\n")

B16 = RowAdd(B15,2,5,5)
print(B16)
print("\n")

[[  1.   0.   5.   7.  -1.  11.  33.]
 [  0.   1.  -2.  -3.   2.  -5. -10.]
 [  0.   0.   1.   0.  -1.   1.   3.]
 [  0.   0. -10. -14.   1. -20. -63.]
 [  0.   0.  -6.  -8.   1. -14. -39.]
 [  0.   0.  -5.  -4.   2.  -7. -24.]]


[[ 0.  0.  2.  0.  0. -1.]
 [ 0.  0. -1.  0.  0.  1.]
 [ 1. -1.  0.  0.  0.  0.]
 [ 0.  0. -4.  1.  0.  1.]
 [ 0.  0. -3.  0.  1.  1.]
 [ 0.  1. -2.  0.  0.  0.]]


[[  1.   0.   0.   7.   4.   6.  18.]
 [  0.   1.  -2.  -3.   2.  -5. -10.]
 [  0.   0.   1.   0.  -1.   1.   3.]
 [  0.   0. -10. -14.   1. -20. -63.]
 [  0.   0.  -6.  -8.   1. -14. -39.]
 [  0.   0.  -5.  -4.   2.  -7. -24.]]


[[-5.  5.  2.  0.  0. -1.]
 [ 0.  0. -1.  0.  0.  1.]
 [ 1. -1.  0.  0.  0.  0.]
 [ 0.  0. -4.  1.  0.  1.]
 [ 0.  0. -3.  0.  1.  1.]
 [ 0.  1. -2.  0.  0.  0.]]


[[  1.   0.   0.   7.   4.   6.  18.]
 [  0.   1.   0.  -3.   0.  -3.  -4.]
 [  0.   0.   1.   0.  -1.   1.   3.]
 [  0.   0. -10. -14.   1. -20. -63.]
 [  0.   0.  -6.  -8.   1. -14. -39.]
 [  0.   0.  -5.  

In [9]:
A17 = RowAdd(A16,4,3,-1)
print(A17)
print("\n")

B17 = RowAdd(B16,4,3,-1)
print(B17)
print("\n")

A18 = RowAdd(A17,5,3,-2)
print(A18)
print("\n")

B18 = RowAdd(B17,5,3,-2)
print(B18)
print("\n")

A19 = RowScale(A18,3,1/2)
print(A19)
print("\n")

B19 = RowScale(B18,3,1/2)
print(B19)
print("\n")

A20 = RowAdd(A19,3,0,-7)
print(A20)
print("\n")

B20 = RowAdd(B19,3,0,-7)
print(B20)
print("\n")

A21 = RowAdd(A20,3,1,3)
print(A21)
print("\n")

B21 = RowAdd(B20,3,1,3)
print(B21)
print("\n")

A22 = RowAdd(A21,3,4,8)
print(A22)
print("\n")

B22 = RowAdd(B21,3,4,8)
print(B22)
print("\n")

A23 = RowAdd(A22,3,5,4)
print(A23)
print("\n")

B23 = RowAdd(B22,3,5,4)
print(B23)
print("\n")

[[  1.   0.   0.   7.   4.   6.  18.]
 [  0.   1.   0.  -3.   0.  -3.  -4.]
 [  0.   0.   1.   0.  -1.   1.   3.]
 [  0.   0.   0.  -6.  -4.  -2. -12.]
 [  0.   0.   0.  -8.  -5.  -8. -21.]
 [  0.   0.   0.  -4.  -3.  -2.  -9.]]


[[-5.  5.  2.  0.  0. -1.]
 [ 2. -2. -1.  0.  0.  1.]
 [ 1. -1.  0.  0.  0.  0.]
 [ 4. -4. -1.  1. -1.  0.]
 [ 6. -6. -3.  0.  1.  1.]
 [ 5. -4. -2.  0.  0.  0.]]


[[  1.   0.   0.   7.   4.   6.  18.]
 [  0.   1.   0.  -3.   0.  -3.  -4.]
 [  0.   0.   1.   0.  -1.   1.   3.]
 [  0.   0.   0.   2.   2.   2.   6.]
 [  0.   0.   0.  -8.  -5.  -8. -21.]
 [  0.   0.   0.  -4.  -3.  -2.  -9.]]


[[-5.  5.  2.  0.  0. -1.]
 [ 2. -2. -1.  0.  0.  1.]
 [ 1. -1.  0.  0.  0.  0.]
 [-6.  4.  3.  1. -1.  0.]
 [ 6. -6. -3.  0.  1.  1.]
 [ 5. -4. -2.  0.  0.  0.]]


[[  1.   0.   0.   7.   4.   6.  18.]
 [  0.   1.   0.  -3.   0.  -3.  -4.]
 [  0.   0.   1.   0.  -1.   1.   3.]
 [  0.   0.   0.   1.   1.   1.   3.]
 [  0.   0.   0.  -8.  -5.  -8. -21.]
 [  0.   0.   0.  

In [10]:
A24 = RowSwap(A23,5,4)
print(A24)
print("\n")

B24 = RowSwap(B23,5,4)
print(B24)
print("\n")

A25 = RowAdd(A24,4,0,3)
print(A25)
print("\n")

B25 = RowAdd(B24,4,0,3)
print(B25)
print("\n")

A26 = RowAdd(A25,4,1,-3)
print(A26)
print("\n")

B26 = RowAdd(B25,4,1,-3)
print(B26)
print("\n")

A27 = RowAdd(A26,4,2,1)
print(A27)
print("\n")

B27 = RowAdd(B26,4,2,1)
print(B27)
print("\n")

A28 = RowAdd(A27,4,3,-1)
print(A28)
print("\n")

B28 = RowAdd(B27,4,3,-1)
print(B28)
print("\n")

A29 = RowAdd(A28,4,5,-3)
print(A29)
print("\n")

B29 = RowAdd(B28,4,5,-3)
print(B29)
print("\n")

[[ 1.  0.  0.  0. -3. -1. -3.]
 [ 0.  1.  0.  0.  3.  0.  5.]
 [ 0.  0.  1.  0. -1.  1.  3.]
 [ 0.  0.  0.  1.  1.  1.  3.]
 [ 0.  0.  0.  0.  1.  2.  3.]
 [ 0.  0.  0.  0.  3.  0.  3.]]


[[ 16.   -9.   -8.5  -3.5   3.5  -1. ]
 [ -7.    4.    3.5   1.5  -1.5   1. ]
 [  1.   -1.    0.    0.    0.    0. ]
 [ -3.    2.    1.5   0.5  -0.5   0. ]
 [ -7.    4.    4.    2.   -2.    0. ]
 [-18.   10.    9.    4.   -3.    1. ]]


[[ 1.  0.  0.  0.  0.  5.  6.]
 [ 0.  1.  0.  0.  3.  0.  5.]
 [ 0.  0.  1.  0. -1.  1.  3.]
 [ 0.  0.  0.  1.  1.  1.  3.]
 [ 0.  0.  0.  0.  1.  2.  3.]
 [ 0.  0.  0.  0.  3.  0.  3.]]


[[ -5.    3.    3.5   2.5  -2.5  -1. ]
 [ -7.    4.    3.5   1.5  -1.5   1. ]
 [  1.   -1.    0.    0.    0.    0. ]
 [ -3.    2.    1.5   0.5  -0.5   0. ]
 [ -7.    4.    4.    2.   -2.    0. ]
 [-18.   10.    9.    4.   -3.    1. ]]


[[ 1.  0.  0.  0.  0.  5.  6.]
 [ 0.  1.  0.  0.  0. -6. -4.]
 [ 0.  0.  1.  0. -1.  1.  3.]
 [ 0.  0.  0.  1.  1.  1.  3.]
 [ 0.  0.  0.  0.  1.  2

In [11]:
A30 = RowScale(A29,5,-1/6)
print(A30)
print("\n")

B30 = RowScale(B29,5,-1/6)
print(B30)
print("\n")

A31 = RowAdd(A30,5,0,-5)
print(A31)
print("\n")

B31 = RowAdd(B30,5,0,-5)
print(B31)
print("\n")

A32 = RowAdd(A31,5,1,6)
print(A32)
print("\n")

B32 = RowAdd(B31,5,1,6)
print(B32)
print("\n")

A33 = RowAdd(A32,5,2,-3)
print(A33)
print("\n")

B33 = RowAdd(B32,5,2,-3)
print(B33)
print("\n")

A34 = RowAdd(A33,5,3,1)
print(A34)
print("\n")

B34 = RowAdd(B33,5,3,1)
print(B34)
print("\n")

A35 = RowAdd(A34,5,4,-2)
print(A35)
print("\n")

B35 = RowAdd(B34,5,4,-2)
print(B35)
print("\n")

print(f"Matrix A = \n{A35}")
print("\n")
print(f"Pembulatan Matrix I = \n{B35.round()}")

[[ 1.  0.  0.  0.  0.  5.  6.]
 [ 0.  1.  0.  0.  0. -6. -4.]
 [ 0.  0.  1.  0.  0.  3.  6.]
 [ 0.  0.  0.  1.  0. -1.  0.]
 [ 0.  0.  0.  0.  1.  2.  3.]
 [-0. -0. -0. -0. -0.  1.  1.]]


[[-5.          3.          3.5         2.5        -2.5        -1.        ]
 [14.         -8.         -8.5        -4.5         4.5         1.        ]
 [-6.          3.          4.          2.         -2.          0.        ]
 [ 4.         -2.         -2.5        -1.5         1.5         0.        ]
 [-7.          4.          4.          2.         -2.          0.        ]
 [-0.5         0.33333333  0.5         0.33333333 -0.5        -0.16666667]]


[[ 1.  0.  0.  0.  0.  0.  1.]
 [ 0.  1.  0.  0.  0. -6. -4.]
 [ 0.  0.  1.  0.  0.  3.  6.]
 [ 0.  0.  0.  1.  0. -1.  0.]
 [ 0.  0.  0.  0.  1.  2.  3.]
 [-0. -0. -0. -0. -0.  1.  1.]]


[[-2.5         1.33333333  1.          0.83333333  0.         -0.16666667]
 [14.         -8.         -8.5        -4.5         4.5         1.        ]
 [-6.          3.  

Jadi ketemu:

x1 = 1

x2 = 2

x3 = 3

x4 = 1

x5 = 1

x6 = 1

Lalu kita substitusikan ke salah satu persamaan. Misal: X_1 + 2X_2 + X_3 + X_4 + 3X_5 + X_6 = 13

Diperoleh => 1 + 2.2 + 1.3 + 1 + 3.1 + 1 = 13

Hasilnya => 13 =13

\begin{align*} MatrixA= \begin{bmatrix} 2&2&2&4&3&6\\ 2&-2&1&4&4&5\\ 1&1&3&4&1&6\\ 3&2&1&1&2&3\\ 2&1&2&3&1&3\\ 1&2&1&1&3&1 \end{bmatrix} \qquad MatrixA^-1= \begin{bmatrix} -2&1&1&1&0&0\\ 11&-6&-6&-2&2&0\\ -4&2&2&1&-0&0\\ 4&-2&-2&-1&1&0\\ 6&3&3&1&-1&0\\ -0&0&0&0&-0&0 \end{bmatrix} \end{align*}
